user/permissions.py

```diff
@@ -0,0 +1,19 @@
+from rest_framework import permissions
+
+
+class IsAdminOrIfAuthenticatedReadOnly(permissions.BasePermission):
+
+    def has_permission(self, request, view):
+        user = request.user
+
+        if user and user.is_staff:
+            return True
+
+        if not user or not user.is_authenticated:
+            return False
+
+        if view.__class__.__name__ == "OrderViewSet":
+            return (request.method in permissions.SAFE_METHODS
+                    or request.method == "POST")
```

Comment:
To będzie czystsze podejście i bardziej pythonowe. Rzadko kiedy chcemy bezpośrednio odwoływać się do atrybutów chronionych
```suggestion
        if isinstance(view, OrderViewSet):
            return request.method in permissions.SAFE_METHODS or request.method == "POST"

```

Reply:
Zobacz -> commit bd148f0dda76fb937bd4ec4633c13ff04c8df1d9

Tam wyjaśnione czemu użyłem takiej formy. Chyba, że istnieje jakiś inny sposób na ominięcie circular import, ale na niego nie wpadłem.

Reply:
Jest na to jeden sposób. Powinien ominąć problem circular importów


```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from cinema.views import OrderViewSet

class IsAdminOrIfAuthenticatedReadOnly(permissions.BasePermission):
    def has_permission(self, request, view):
        user = request.user
        if user and user.is_staff:
            return True
        if not user or not user.is_authenticated:
            return False
        if isinstance(view, OrderViewSet):
            return request.method in permissions.SAFE_METHODS or request.method == "POST"
        return request.method in permissions.SAFE_METHODS
```

Reply:
Spróbuj tego i daj znać, czy śmiga

Reply:
Nie śmiga. To rozwiązanie powodowało 17 błędów "NameError: name 'OrderViewSet' is not defined" w testach.
Znalazłem inne rozwiązanie przenosząc import do środka metody has_permission.